chronos/asyncfutures2.nim

```diff
@@ -613,18 +613,23 @@ macro checkFutureExceptions*(f, typ: typed): untyped =
     quote do: raiseAssert("Unhandled future exception: " & `f`.error.msg)
   )
 
-proc read*[T](future: Future[T] ): T {.
-     raises: [Defect, CatchableError].} =
+# we have to rely on inference here
+{.pop.}
+proc read*(future: Future | RaiseTrackingFuture ): auto =
```

Comment:
just make a separate read function for RTF? the `auto` return type here is unnecessarily wide as well, there's too many compromises for too little benefit

Reply:
Indeed, we can avoid it here by making a specialized read for RTF